src/main/webapp/app/shared/components/code-button/code-button.component.ts

```diff
@@ -79,27 +80,29 @@ export class CodeButtonComponent implements OnInit, OnChanges {
         private ideSettingsService: IdeSettingsService,
     ) {}
 
-    ngOnInit() {
-        this.accountService
-            .identity()
-            .then((user) => {
-                this.user = user!;
-                this.refreshTokenState();
+    async ngOnInit() {
+        this.user = (await this.accountService.identity())!;
```

Comment:
does this change really improve the code? We have the other pattern with 
```
.identity()
             .then((user) => ...
```

many times in the code. I don't think we should have both solutions except there is a really good reason!

Reply:
The main idea behind this was to make the code less nested, and to make it simpler to see what's going on. Generally I think the .identify().then(..) makes sense, but in the code button are so many things going on, that it's very easy to get lost

Reply:
https://github.com/ls1intum/Artemis/pull/9696 changes this for a bug fix